crates/supervisor/core/src/config/rollup_config_set.rs

```diff
@@ -53,14 +53,22 @@ impl RollupConfig {
     }
 
     /// Creates a new [`RollupConfig`] with the given genesis and block time.
-    pub fn new_from_rollup_config(config: kona_genesis::RollupConfig, l1_block: BlockInfo) -> Self {
-        // todo: cross check the l1_block with the config
-        // issue: https://github.com/op-rs/kona/issues/2254
-        Self {
+    pub fn new_from_rollup_config(
+        config: kona_genesis::RollupConfig,
+        l1_block: BlockInfo,
+    ) -> Result<Self, String> {
+        if config.genesis.l1.number != l1_block.number {
+            return Err(format!(
+                "L1 block number mismatch: expected {}, but got {}",
+                config.genesis.l1.number, l1_block.number
+            ));
```

Comment:
please make this into a proper error variant of most appropriate error make or new error enum if necessary, and avoid using string